doc/src/explanation/glossary.md

```diff
@@ -0,0 +1,712 @@
+    SymPy has several internal classes to represent matrices, depending on
+    whether the matrix is symbolic ({class}`~.MatrixExpr`) or explicit,
+    mutable or immutable, dense or sparse, and what type the underlying
+    elements are, but these are often all just called "Matrix".
+
+mpmath
+
+    [*mpmath*](https://mpmath.org/) is a pure Python library for arbitrary
+    precision numerics. It is a [hard dependency](dependencies-mpmath) of
+    SymPy. mpmath is capable of computing {term}`numerical <numeric>`
+    functions to any given number of digits. mpmath is used under the hood
+    whenever SymPy evaluates an expression numerically, such as when using
+    {term}`evalf`.
+
+Numeric
+
+    A *numeric* representation or algorithm is one that operates directly on
+    numeric inputs. It is in contrast with a *{term}`symbolic`* representation
+    or algorithm, which can work with objects in an unevaluated form. Often a
+    numerical algorithm is quite different from a symbolic one. For example,
+    numerically solving an ODE typically means evaluating the ODE using an
+    algorithm like
+    [Runge–Kutta](https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta_methods)
+    to find a set of numeric points given an initial condition, whereas
+    symbolically solving an ODE (such as with SymPy's {func}`~.dsolve`) means
+    mathematically manipulating the ODE to produce a {term}`symbolic`
+    {term}`equation` that represents the solution. A symbolic ODE solution may
+    including symbolic constants which can represent any numerical value.
+    Numeric algorithms are typically designed around issues caused by
+    floating-point numbers such as loss of precision and numerical stability,
+    whereas symbolic algorithms are not concerned with these things because
+    they compute things exactly.
+
+    Most scientific libraries other than SymPy, such as NumPy or SciPy, are
+    strictly numerical, meaning the functions in those libraries can only
+    operate on specific numeric inputs. They will not work with SymPy
+    expressions, because their algorithms are not designed to work with
+    symbolic inputs. SymPy focuses on symbolic functions, leaving purely
+    numerical code to other tools like NumPy. However, SymPy does interface
+    with numerical libraries via tools like {term}`code generation` and
+    {term}`lambdify()`.
+
+Number
+
+    *Number* can refer to two things in SymPy:
+
+    - The class {class}`~.Number`, which is the base class for explicit
+      numbers ({class}`~.Integer`, {class}`~.Rational`, and {class}`~.Float`).
+      Symbolic numeric constants like {class}`pi <sympy.core.numbers.Pi>` are
+      not instances of `Number`.
+
+    - Lowercase "*number*", as in the `is_number` property, refers to any
+      {term}`expression` that can be {term}`evalfed <evalf>` into an explicit
+      `Number`. This includes symbolic constants like {class}`pi
```

Comment:
It's probably worth mentioning symbols here. Most of the time the distinction is about not having free symbols although it is also about distinguishing a number from something like a matrix.

Reply:
I'm really not clear what you think I should add here. Symbols are not numbers, according to this definition (they don't represent a specific numerical value).